### PR TITLE
Refactor generators database config

### PR DESCRIPTION
**Pull request by brennovich**
- Extract common behaviour between generators (app, container)
- Add unit specs covering generators'  database config logic
- Add multi-platform support (mri, jruby)
- Normalize behaviour of database configuration between App and Container generators (add `.sqlite` into database name when SQLite engine)

I'm aware of #324, but I'm taking a different approach, instead dumping all database config logic into `Abstract` class (where I believe we should only define interface, not logic) I extracted to a new class `DatabaseConfig`. It think this should be incorporated in #324

### CODE REVIEW

**Comment by joneslee85** [comment]
@brennovich I love this approach, it makes senses as it is much easier to work with other platforms variations. Whenever you have time, ping me on chat, I want to pair with you to review this.

File: lib/lotus/generators/database_config.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
+module Lotus
+  module Generators
+    class DatabaseConfig
+      SUPPORTED_ENGINES = {
+        'mysql'      => { type: :sql,         mri: 'mysql',   jruby: 'jdbc-mysql'    },
```
Comment (joneslee85) [inline lib/lotus/generators/database_config.rb]: I don't think we should support the old `mysql` driver anymore, can you please change it to `mysql2` for MRI?

Comment (joneslee85) [inline lib/lotus/generators/database_config.rb]: no, on second look, it seems to me we do support that, please ignore my comment above

File: test/integration/cli/generate_test.rb [hunk 2]
```diff
@@ -13,7 +14,7 @@
 
     def create_temporary_dir
       @tmp = Pathname.new(@pwd = Dir.pwd).join('tmp/integration/cli/generate')
-      @tmp.rmtree if @tmp.exist?
+      FileUtils.rm_rf(@tmp) if @tmp.exist?
```
Comment (joneslee85) [inline test/integration/cli/generate_test.rb]: is there value of checking `@tmp.exist?` anymore?

File: lib/lotus/generators/database_config.rb (hunk 1 above)
Comment (brennovich) [inline lib/lotus/generators/database_config.rb]: Maybe we should drop support, I believe that not even Rails supports this

**Comment by brennovich** [comment]
@joneslee85 sure! Any preferred service (hangouts, Skype, irc, etc)?

**Comment by joneslee85** [comment]
@brennovich use our gitter.im chat room, http://chat.lotusrb.org

File: test/integration/cli/generate_test.rb (hunk 2 above)
Comment (brennovich) [inline test/integration/cli/generate_test.rb]: You're right :)

Dropped 0af5573

**Comment by carloslopes** [comment]
Awesome! :heart:

**Comment by joneslee85** [comment]
Looks good, please squash into 1

cc @lotus/core

File: lib/lotus/generators/database_config.rb (hunk 1 above)
Comment (joneslee85) [inline lib/lotus/generators/database_config.rb]: @brennovich please do it in a separate PR instead

File: lib/lotus/generators/database_config.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        when 'mysql2'
+          "mysql2://localhost/#{ name }"
+        when 'postgresql', 'postgres'
+          "postgres://localhost/#{ name }"
+        when 'sqlite', 'sqlite3'
+          "sqlite://db/#{ Shellwords.escape(name) }"
+        when 'memory'
+          "memory://localhost/#{ name }"
+        when 'filesystem'
+          "file:///db/#{ name }"
```
Comment (joneslee85) [inline lib/lotus/generators/database_config.rb]: please also do `Shellwords.escape` here

Comment (brennovich) [inline lib/lotus/generators/database_config.rb]: Done https://github.com/brennovich/lotus/commit/0dd955929d9d0f3d49ccc2f38ecfdbf48ef1c2e7#diff-38afc003617f0593d1a62c002f6a8f42R74

**Comment by brennovich** [comment]
Squashed :)
